book/garbage-collector/README.md

```diff
@@ -624,24 +610,15 @@ duplicate some immutable values such as floating-point values in arrays.
 These may be finalized while another duplicate copy is being used by the
 program.
 
-For this reason, attach finalizers only to values that you are
-explicitly sure are heap-allocated and aren't immutable. A common use
-is to attach them to file descriptors to ensure they are closed.  <!--
-TODO: This doesn't really line up, since file descriptors are not
-heap-allocated values in OCaml. --> However, the finalizer normally
-shouldn't be the primary way of closing the file descriptor, since it
-depends on the GC running in order to collect the value. For a busy
-system, you can easily run out of a scarce resource such as file
-descriptors before the GC catches up.
+For this reason, attach finalizers only to values that you are explicitly
+sure are boxed and heap-allocated and aren't immutable.
 :::
 
 Core provides a `Heap_block` module that dynamically checks if a given
 value is suitable for finalizing.  Core keeps the functions for
-registering finalizers in the `Expert` module finalizers can be pretty
-hard to reason about in multi-threaded contexts, since finalizers can
-run at any time in any thread.
-[heaps/Heap_block module]{.idx}
-
+registering finalizers in the `Core.Gc.Expert` module as finalizers
+can be pretty hard to reason about in multi-threaded contexts, since finalizers
+can run at any time in any thread.  [heaps/Heap_block module]{.idx}
```

Comment:
This seems like a bit of a run-on sentence.

Reply:
Maybe, "Core keeps the functions for registering finalizers in the Core.Gc.Expert module, since finalizers can be pretty hard to reason about, particularly in multi-threaded contexts, where finalizers can run at any time in any thread.

Reply:
pushed a better reword